spec/datadog/tracing/integration_spec.rb

```diff
@@ -24,9 +24,31 @@
   shared_context 'agent-based test' do
     before do
       skip unless ENV['TEST_DATADOG_INTEGRATION']
+
+      # Ensure background Writer worker doesn't wait, making tests faster.
+      stub_const('Datadog::Tracing::Workers::AsyncTransport::DEFAULT_FLUSH_INTERVAL', 0)
+
+      # Capture trace segments as they are about to be serialized
+      segments = trace_segments
+      allow_any_instance_of(Datadog::Transport::TraceFormatter)
+        .to receive(:format!).and_wrap_original do |original|
+          segments << original.call
+        end
```

Comment:
I know this was already like this but... is it just me or is this super wonky? Could we use webmock or something similar to look at what comes out of the pipe without interceding in it?

Reply:
Webmock has it serialized as msgpack. Maybe I can try to deserialize it from there. 🤔